api/INDEXING.md

```diff
@@ -0,0 +1,80 @@
+# Indexing
+* [Overview](#overview)
+* [Running Indexing Jobs](#running-indexing-jobs)
+    * [Expand Underlay Config](#expand-underlay-config)
+    * [Create Index Dataset](#create-index-dataset)
+    * [Kickoff Jobs](#kickoff-jobs)
+
+## Overview
+Each underlay config specifies the mapping from the source data to Tanagra's entity model. Tanagra can query the
+source data directly, but for improved performance, Tanagra generates indexed tables and queries those instead.
+Each underlay config also specifies a data pointer where Tanagra can write indexed tables to.
+
+Generating the index tables is part of the deployment process; It is not managed by the service. There is a basic
+command line interface to run the indexing jobs. Currently, this CLI just uses Gradle's application plugin, so the
+commands are actually Gradle commands. In the future, we should improve this UX (e.g. package a proper CLI, add 
+service endpoints for kicking off and monitoring these jobs, etc.).
+
+## Running Indexing Jobs
+There are 3 steps to generating the index tables:
+1. Expand the user-specified underlay config to include information from scanning the source data. For example,
+data types and UI hints.
+2. Create the index dataset, if it doesn't already exist.
+3. Kickoff jobs for each entity and entity group.
+
+### Expand Underlay Config
+Set the default application credentials to a service account key file that has read access to both the source and 
+index data.
+```
+export GOOGLE_APPLICATION_CREDENTIALS=/credentials/indexing_sa.json
```

Comment:
The instructions are not the same for all sections. For expanding the config and running indexing jobs, you need to define the env var. For creating the BQ dataset, you need to configure the `gcloud`/`bq` CLI.

Reply:
I thought all the methods of setting the default application credentials were interchangeable? Does neither of the two methods work for all three cases?

Reply:
For the `EXPAND_CONFIG` and `INDEX_ENTITY/GROUP` commands, we pull the default application credentials. You can set this several different ways, including with the env var as I've indicated here. The `gcloud` CLI family (incl `bq`) does not use these for most of their own commands. You need to authenticate directly.

One way to address this is to move the creation of the index dataset into the Tanagra codebase instead of asking users to do it with `bq` or Cloud Console. This might be the way to go, since a companion "delete index dataset" command would be useful for testing.